### PR TITLE
Add example manifests and provisioners

### PR DESCRIPTION
**Pull request by rothgar**
**2. Description of changes:**
Adds some example manifests and provisioners that I've been using for demos and talks. I think they'll be helpful for people learning how to use Karpenter.

**3. How was this change tested?**
running the examples in talks and testing.

**4. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [x] No

We might want to add docs that link to the examples at some point but don't need to.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/provisioner/provisioner-100-cpu.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+apiVersion: karpenter.sh/v1alpha5
+kind: Provisioner
+metadata:
+  name: default
+spec:
+  architectures:
```
Comment (ellistarn) [inline examples/provisioner/provisioner-100-cpu.yaml:6]: Did you test this spec? This looks like v1alpha3 :P

File: examples/workloads/spot.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    spec:
+      containers:
+      - image: public.ecr.aws/eks-distro/kubernetes/pause:3.2
+        name: inflate-spot
+        resources:
+          requests:
+            cpu: "1"
+            memory: 256M
+      nodeSelector:
+        node.k8s.aws/capacity-type: spot
```
Comment (bwagner5) [inline examples/workloads/spot.yaml:23]: this is now `karpenter.sh/capacity-type`

File: examples/provisioner/provisioner-100-cpu.yaml (hunk 1 above)
Comment (rothgar) [reply examples/provisioner/provisioner-100-cpu.yaml:6]: I tested all the workloads and _most_ of the provisioners. :grimacing: Let me double check on these.

**Review by bwagner5 — APPROVED** [review]
Nice! Thanks!

**Pull request merged**